create is_in_the_same_node on cpu

helunwencser opened the pull request:
Summary: `in_the_same_node_as` is only called on the host. This PR makes `is_in_the_same_node` to be created on the host directly; otherwise, this function fails if `torch.set_default_device()` is called during device initialization because `is_in_the_same_node` will be created on the device by default.

Test Plan:
Before:
vLLM fails to initialize distributed group due to the following error:
```RuntimeError: No backend type associated with device type XXX```

After:
vLLM can initialize distributed group successfully

Differential Revision: D84630888

houseroad reviewed (commented):
Looks good. 

Can we fix the linter (pre-commit)?

Also can we add a unittest?

Commit:
create is_in_the_same_node on cpu (#26832)

Summary:

`in_the_same_node_as` is only called on the host. This PR makes `is_in_the_same_node` to be created on the host directly; otherwise, this function fails if `torch.set_default_device()` is called during device initialization because `is_in_the_same_node` will be created on the device by default.

Signed-off-by: Lunwen He <lunwenh@meta.com>

Test Plan:
Before:
vLLM fails to initialize distributed group on MTIA due to P1991634642.
After:
vLLM can initialize distributed group on MTIA successfully

Differential Revision: D84630888
